core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -193,24 +147,24 @@ public void dispatchRequest(final RestRequest request, final RestChannel channel
                     threadContext.putHeader(key, httpHeader);
                 }
             }
-            if (filters.length == 0) {
-                executeHandler(request, channel, client);
+
+            final RestHandler rawHandler = getHandler(request);
+            final RestHandler handler = handlerWrapper.apply(rawHandler);
+
+            if (handler != null) {
```

Comment:
should we allow null here? I think it must not return null but the raw handler instead? maybe I am missing something